lang/py/src/avro/io.py

```diff
@@ -341,6 +389,71 @@ def write_double(self, datum):
     self.write(chr((bits >> 48) & 0xFF))
     self.write(chr((bits >> 56) & 0xFF))
 
+  def write_decimal_bytes(self, datum, scale):
+    """
+    Decimal in bytes are encoded as long. Since size of packed value in bytes for
+    signed long is 8, 8 bytes are written.
+    """
+    sign, digits, exp = datum.as_tuple()
+    if exp > scale:
+      raise AvroTypeException('Scale provided in schema does not match the decimal')
+
+    unscaled_datum = 0
+    for digit in digits:
+      unscaled_datum = (unscaled_datum * 10) + digit
+
+    bits_req = unscaled_datum.bit_length()
+    if sign:
```

Comment:
Hey @mtth , the last test case will work cause of [this](https://github.com/premsantosh/avro/blob/master/lang/py/src/avro/io.py#L398). As for the others, am not able to understand how you got about getting those values. I have tried encoding the exact same values you have commented out and then decoding those using the java decoder and they seem to work.
![screen shot 2016-05-11 at 12 07 58 pm](https://cloud.githubusercontent.com/assets/1332635/15193125/33e84286-1771-11e6-8459-33fbcbb3e25c.png)
![screen shot 2016-05-11 at 12 10 17 pm](https://cloud.githubusercontent.com/assets/1332635/15193168/5beea446-1771-11e6-89b8-dae25cd7284a.png)
`\x04\xff\x80` produces `1001111111110000000`. The `read_long()` then reads the `100` and knows it has to read 2 bytes. Then the `read_decimal_from_bytes()` is passed in a size of 2 bytes to read and it reads `1111111110000000` which translates to `-128`.
The last test (`write_datum(Decimal('1'), writers_schema)`) shouldn't work with the schema provided cause the precision and scale do not match. 
Could you please elaborate on your thoughts?